src/git.rs

```diff
@@ -112,6 +116,21 @@ pub fn yank(app: &App, krate: &str, version: &semver::Version, yanked: bool) ->
     })
 }
 
+/// Commits and pushes to the crates.io index.
+///
+/// There are currently 2 instances of the crates.io backend running
+/// on Heroku, and they race against each other e.g. if 2 pushes occur,
+/// then one will succeed while the other will need to be rebased before
+/// being pushed.
+///
+/// A maximum of 20 attempts to commit and push to the index currently
+/// accounts for the amount of traffic publishing crates, though this will
+/// eventually need to be changed past a certain point.
+///
+/// Notes:
+/// Currently, this function is called from the HTTP thread and is blocking.
+/// This could be changed to run from a different thread and use a callback
+/// upon completion to the HTTP thread.
```

Comment:
I wouldn't say that it would callback to the HTTP thread, since that means the HTTP thread would still be waiting. The point of spawning a thread would be that the HTTP request could return a response and be finished without having to wait for the git push to complete (or uploading to S3, or rendering the README, etc). That would mean we wouldn't return success or failure to `cargo publish` at all, we'd say something like "publish in progress, check [something] for status" and we'd have to figure out how we want the user to be able to check for status (run `cargo publish --status`, check in crates.io's UI, we send them an email, etc).